common/src/main/java/net/vyhub/abstractClasses/ABans.java

```diff
@@ -0,0 +1,264 @@
+package net.vyhub.abstractClasses;
+
+import com.google.gson.reflect.TypeToken;
+import net.vyhub.VyHubPlatform;
+import net.vyhub.entity.Ban;
+import net.vyhub.entity.MinecraftBan;
+import net.vyhub.entity.VyHubUser;
+import net.vyhub.lib.Cache;
+import net.vyhub.lib.Utility;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import retrofit2.Response;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+
+import static java.util.logging.Level.*;
+import static net.vyhub.VyHubAPI.gson;
+
+public abstract class ABans {
+    public static DateTimeFormatter isoDateFormatter = DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+    public static DateTimeFormatter mcDateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss ZZZ");
+
+    private static Set<String> processedPlayers = new HashSet<>();
+
+    private static Map<String, MinecraftBan> minecraftBans = null;
+    private static Map<String, List<Ban>> vyhubBans = null;
+
+    private final VyHubPlatform platform;
+    private final AUser aUser;
+    private final AGroups aGroups;
+
+    public ABans(VyHubPlatform platform, AUser aUser, AGroups aGroups) {
+        this.platform = platform;
+        this.aUser = aUser;
+        this.aGroups = aGroups;
+    }
+
+    public VyHubPlatform getPlatform() {
+        return platform;
+    }
```

Comment:
As every abstract class has the platform and some have the getPlatform method, maybe inherit from a parent class to avoid redundancy?